From a3e4135c543be8d3a054e6f74629240370d111ed Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Mon, 27 May 2019 15:55:19 +0800
Subject: [PATCH] policy/modules/system/sysnetwork: allow ifconfig_t to read
 dhcp client state files

Fixes:
type=AVC msg=audit(1558942740.789:50): avc:  denied  { read } for
pid=221 comm="ip" path="/var/lib/dhcp/dhclient.leases" dev="vda"
ino=29858 scontext=system_u:system_r:ifconfig_t
tcontext=system_u:object_r:dhcpc_state_t tclass=file permissive=1

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/sysnetwork.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/sysnetwork.te b/policy/modules/system/sysnetwork.te
index bbdbcdc7e..a77738924 100644
--- a/policy/modules/system/sysnetwork.te
+++ b/policy/modules/system/sysnetwork.te
@@ -319,6 +319,8 @@ kernel_request_load_module(ifconfig_t)
 kernel_search_network_sysctl(ifconfig_t)
 kernel_rw_net_sysctls(ifconfig_t)
 
+sysnet_read_dhcpc_state(ifconfig_t)
+
 corenet_rw_tun_tap_dev(ifconfig_t)
 
 dev_read_sysfs(ifconfig_t)
-- 
2.17.1

